src/design/crosslink/construction.md

```diff
@@ -99,6 +101,12 @@ It may be the case that a ballot cast for $P$ is not in honest view when it is u
 There may be multiple distinct ballots or distinct ballot messages attempting to cast a given voting unit for the same proposal; this is undesirable for bandwidth usage, but it is not necessary to consider it to be non‑honest behaviour for the purpose of security analysis, as long as such ballots are not double‑counted toward the two‑thirds threshold.
 ```
 
+```admonish warning "Security caveat"
+The **one‑third bound on non‑honest voting** property considers all ballots cast in the entire execution. In particular, it is possible that a validator’s key is compromised and then used to cast its voting units for a proposal of an epoch long finished. If the number of voting units cast non-honestly for any epoch *ever* reaches one third of the total voting units for that epoch during an execution, then the **one‑third bound on non‑honest voting** property is violated for that execution.
+
+Therefore, validator keys of honest nodes must remain secret indefinitely. Whenever a key is rotated, the old key must be securely deleted. For further discussion and potential improvements, see [tfl-book issue #140](https://github.com/Electric-Coin-Company/tfl-book/issues/140).
```

Comment:
+1

If I recall correctly, there's a conventional name for the attack of using compromised keys on long past epochs (and I think it is distinct from long-range attacks, IIRC). I'll look that up.

Reply:
It was from this video: [A Survey of Proof-of-Stake (PoS) Blockchain Designs with Valeria Nikolaenko | a16z crypto research](https://www.youtube.com/watch?v=mZ-Ya7NRDxM)

Reply:
Linked the video here for future reference: https://www.youtube.com/watch?v=-uxHoEfxXC4